File: backend/internal/expression/identifier.go
package expression

import (
	"bytes"

	"app/internal/token"
)

type Identifier struct {
	name token.Token
}

func NewIdentifier(name token.Token) *Identifier {
	return &Identifier{
		name: name,
	}
}

func (ident *Identifier) DebugString() string {
	var buffer bytes.Buffer
	buffer.WriteString("IDENT(")
	buffer.WriteString(ident.name.Literal)
	buffer.WriteString(")")
	return buffer.String()
}

func (ident *Identifier) PrettyString() string {
	return ident.name.Literal
}

File: backend/internal/expressions/identifier.py
from backend.internal.expressions import Expression
from backend.internal.tokens import Token


class Identifier(Expression):
    def __init__(self, name: Token) -> None:
        super().__init__()
        self.name = name

    def __eq__(self, other) -> bool:
        return isinstance(other, Identifier) and self.name == other.name

    def __repr__(self) -> str:
        return f"IDENT({repr(self.name)})"

    def pretty_str(self) -> str:
        return self.name.literal
